testsuite/MDAnalysisTests/coordinates/test_h5md.py

```diff
@@ -894,3 +895,20 @@ def test_writer_no_h5py(self, Writer, outfile):
                         u.atoms.n_atoms) as W:
                 for ts in u.trajectory:
                     W.write(universe)
+
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+class TestH5MDReaderWithObservables(object):
+    """Read H5MD file with 'observables/atoms/energy'."""
+
+    prec = 3
+    ext = 'h5md'
+
+    @pytest.fixture(scope='class')
```

Comment:
I may be mistaken, but I think the `test_n_frames` method isn't testing the issue- if the reader was going to except out due to a key error, this would happen during the line in the `universe` fixture:
```
reader = H5MDReader(H5MD_energy, convert_units=True)
```
when the H5MDReader's `__init__` is called and the first call to `_read_next_timestep` is made. The test method doesn't call any code that reads a frame, it just ends up calling the reader's `n_frames` method. I agree with your approach to fixing the issue, just maybe have the test initialize or initialize and iterate through the reader so that the potential issue will occur in the scope of the test

Reply:
Also make sure to update the CHANGELOG!

Reply:
Should be fixed now. The CI failing seems not related to this PR?

Reply:
Small nitpick- can you change the name of the test to make it represent what it is testing? Also, the assert statement seems unrelated to what the test is testing. Maybe write an except and `pytest.fail` statement upon raising a KeyError but otherwise don't have an assert statement. Here's an example from `testsuite/coordinates/base.py`:
```
    def test_frame_jump_issue1942(self, ref, reader):
        """Test for issue 1942 (especially XDR on macOS)"""
        reader.rewind()
        try:
            for ii in range(ref.n_frames + 2):
                reader[0]
        except StopIteration:
            pytest.fail("Frame-seeking wrongly iterated (#1942)")
```
Otherwise LGTM

Reply:
I updated the test and used `pytest.fail`. I hope the test structure now is a bit cleaner. Thanks for the review.